### PR TITLE
Translation script

### PR DESCRIPTION
**Pull request by jkokh**

### CODE REVIEW

**Review by EGreg — CHANGES_REQUESTED** [review]
This is good, but please rename the files to `platform/classes/Q/Translate.php` and `platform/classes/Q/Translate/*.php` for the various adapters. Rename the classes to be named `Q_Translate` and `Q_Translate_*`.

And rename "Auto" one to "Q_Translate_Google". Also change "format" to be either "google" or "human", and rename "format" option to be called "translator".

In addition, please allow the following options syntax:

`--in <dir> [format]`
`--out <dir> [format]`

the [format] would default to "Q/text" but may later be other adapters such as "Xcode/translate" and "Streams/interests" etc. This would result in the inputs being parsed differently, or outputs being formatted differently.

Where do we store these adapters for input formats and output formats? Let's call the base class `Q_Translate_Format` with static methods, and then load adapter `Q_Translate_Format_Q_Text` stored in `classes/Q/Translate/Format/Q/Text.php`. Later we may add other adapters. These adapters will have static methods `load` and `save`.

So to summarize: We will now have the following classes, loaded on demand and used for translation by any PHP code:

`Q_Translate` and its adapters
`Q_Translate_Format` and its adapters.

**Comment by EGreg** [comment]
I can accept this pull request after you implement what was in the last comment, starting with

> This is good, but please rename the files to platform/classes/Q/Translate.php and platform/classes/Q/Translate/*.php for the various adapters. Rename the classes to be named Q_Translate and Q_Translate_*.

**Review by EGreg — CHANGES_REQUESTED** [review]
Need changes from comment which I left before, to create `classes/Translate/*` and much more.

**Comment by EGreg** [comment]
Did we finally get this refactoring done? Please update this branch and the pull request. I want `classes/Q/Translate.php` and other classes!
